client/src/components/EditAssociatedPositionsModal.js

```diff
@@ -183,7 +183,7 @@ const BaseEditAssociatedPositionsModal = ({
   }
 
   function save(values, form) {
-    const newPosition = new Position(position)
+    const newPosition = Object.without(new Position(values), "notes")
```

Comment:
I've changed Object.without(new Position(values), "notes") to new Position(Object.without(values, "notes")), and actually also changed other similar expressions to the more efficient way.

Reply:
Which I will have to change, as it is not ok in all cases.

Reply:
@gjvoosten I prefer to keep it as
`const newPosition = Object.without(new Position(values), "notes")`

as this would result in newPosition having no notes key.

If I would use 
`const newPosition = new Position(Object.without(values, "notes"))`
newPosition would have an empty array as notes property.

If we keep it as it is now we know for sure that we don't reset the notes for instance.

What's strange is that if I revert the change, so if I don't remove the notes, I no longer get the error I used to. At least I can't reproduce it any more.

Reply:
But would there really be a difference between sending a Position to the server with notes `null` or `[]`?

Reply:
Actually, I think we don't even send null,, we just don't send the notes.
I should check on the sever side.

Reply:
Let me rephrase: is there really a difference if this Position object on the server has `null` notes or an empty list? Because that is what matters.

Reply:
If the Position has notes, and we do afterwards a mutation in which we set notes to [], it "might mean" for the server "remove all notes". And if we don't provide the notes at all it "might mean" don't update the notes.

Reply:
I've tested it and it looks like if we try to update a position already having notes and we provide notes=[] in the mutation, the notes are not being deleted.

So I could change it to
`const newPosition = new Position(Object.without(values, "notes"))`
but do you find the change really important?

If I change that, for consistency I should also change things like:
```
    const person = Object.without(
      new Person(values),
      "notes",
      "firstName",
      "lastName",
      "customFields", // initial JSON from the db
      DEFAULT_CUSTOM_FIELDS_PARENT
    )
```
to
```
    const person = Object.without(
      new Person(Object.without(values, "notes")),
      "firstName",
      "lastName",
      "customFields", // initial JSON from the db
      DEFAULT_CUSTOM_FIELDS_PARENT
    )
```
and this might be strange for someone reading this code.

Reply:
It's not important, you can leave it as it was.